nilearn/plotting/tests/test_img_plotting.py

```diff
@@ -379,3 +379,11 @@ def test_plot_connectome_exceptions():
                         adjacency_matrix, node_coords,
                         node_kwargs={'c': 'blue'},
                         **kwargs)
+
+
+def test_singleton_ax_dim():
+    for axis, direction in enumerate("xyz"):
+        shape = [5, 6, 7]
+        shape[axis] = 1
+        img = nibabel.Nifti1Image(np.ones(shape), np.eye(4))
+        plot_stat_map(img, None, display_mode=direction)
```

Comment:
How much time is this adding to the tests?

Unfortunately our plotting tests are a bit long, given that they involve a lot of matplotlib machinery. Thus we need to be careful here, and sometimes we will not test the full plotting function, but only the internal cutting one, if the cost in time is too much. Indeed, if the tests start taking too long to run, people won't run them anymore.